### PR TITLE
Extend Plugin-System for CLI Plugins

### PR DESCRIPTION
**Pull request by chris-rock**

### CODE REVIEW

File: lib/inspec/plugins.rb [hunk 1]
```diff
@@ -3,10 +3,12 @@
 # author: Christoph Hartmann
 
 require 'forwardable'
-
+puts 'plugins'
```
Comment (srenatus) [inline lib/inspec/plugins.rb]: a left-over, I suppose :)

Comment (chris-rock) [inline lib/inspec/plugins.rb]: thanks @srenatus , missed that!

File: bin/inspec [hunk 2]
```diff
@@ -251,4 +251,17 @@ class InspecCLI < Thor # rubocop:disable Metrics/ClassLength
...
+
+# load CLI plugins before the Inspec CLI has been started
+Inspec::Plugins::CLI.registry.each { |_subcommand, params|
+  Inspec::InspecCLI.register(
+    params[:klass],
+    params[:subcommand_name],
+    params[:usage],
+    params[:description],
+    params[:options],
+  )
```
Comment (arlimus) [inline bin/inspec:263]: nice!! :D

File: lib/inspec.rb [hunk 3]
```diff
@@ -11,11 +11,13 @@
...
 require 'inspec/version'
-require 'inspec/plugins'
 require 'inspec/profile'
-require 'inspec/resource'
 require 'inspec/rspec_json_formatter'
 require 'inspec/rule'
+require 'matchers/matchers'
 require 'inspec/runner'
+# ensure resource and plugins are loaded after runner, because the runner loads
+# targets
```
Comment (arlimus) [inline lib/inspec.rb:20]: great catch!

**Pull request merged**